public/extensions/door.js

```diff
@@ -0,0 +1,127 @@
+import daily from "./core.js";
+import Socket from "./socketiostate.js";
+
+let state, call;
+let socket;
+let pounced = false;
+let openCallback, closeCallback, onceOpenCallback;
+let openCallbacks = [];
+let closeCallbacks = [];
+
+/* Defaults */
+let props = {
+  room: "", // used to build the presence socket namespace key.
+  domain: "", // used to build the presence socket namespace key.
+};
+let key = "door";
+
+let buttons = {
+  closeDoor: {
+    iconPath: "https://www.svgrepo.com/show/344140/door-open-fill.svg",
+    label: "Close Door",
+    tooltip: "Close Door",
+  },
+  openDoor: {
+    iconPath: "https://www.svgrepo.com/show/344139/door-closed-fill.svg",
+    label: "Open Door",
+    tooltip: "Open Door",
+  },
+  toggleDoor: {
+    iconPath: "https://www.svgrepo.com/show/344139/door-closed-fill.svg",
+    label: "Toggle Door",
+    tooltip: "Toggle Door",
+  },
+};
+
+/* Public interface */
+let self;
+export default self = {
+  connect: function (p) {
+    props = p;
+    let key = `${p.domain}/${p.room}/door`;
+    socket = new Socket({ key });
+    socket.onStateUpdate((s) => {
+      state = s;
+      if (state.clients == 1 && !state["doorState"]) {
+        // nobody's here yet; open the door
+        self.open();
+      } else if (state["doorState"] === "open") {
```

Comment:
I know we use this pattern a whole lot in pluot-core, and at times it bothers me. :) It feels kind of silly to put `const OPEN = "open"` and then use OPEN all over the place, because if we ever need to change it, it's a slightly different "find and replace in project." :) But then as I was typing that, I realized it _would_ actually be useful to do `DOOR_OPEN` and `DOOR_CLOSED` instead, because those are more easily find-and-replace-able.